cli/mender-artifact/copy.go

```diff
@@ -261,3 +275,44 @@ func parseCLIOptions(c *cli.Context) int {
 		return parseError
 	}
 }
+
+// createTmpFileWithPerm Takes a file, and creates a temp-file copy of the
+// current file, with the permissions given by perm.
+func createTmpFileWithPerm(f *os.File, perm os.FileMode) (string, error) {
```

Comment:
An slightly cleaner interface can be `createTmpFileWithPerm` taking just the filename (aka `f.Name()` and the permissions, then returning a filename (as it is doing) instead of having a full file object as the input, an string as the output.

Reply:
Agreed!

Reply:
On second thoughts, I will leave this, as the file contents are also copied over here, thus passing a string will mean, closing the file. Passing the string, then re-opening it in the function.

Reply:
Oh, you are right sir!